pkg/ccl/backupccl/backup_test.go

```diff
@@ -5022,80 +5022,110 @@ func TestProtectedTimestampsDuringBackup(t *testing.T) {
 	runner := sqlutils.MakeSQLRunner(conn)
 	runner.Exec(t, "CREATE TABLE foo (k INT PRIMARY KEY, v BYTES)")
 	close(allowRequest)
-	runner.Exec(t, `BACKUP TABLE FOO TO 'nodelocal://0/foo'`) // create a base backup.
-	allowRequest = make(chan struct{})
-	runner.Exec(t, "SET CLUSTER SETTING kv.protectedts.poll_interval = '100ms';")
-	runner.Exec(t, "ALTER TABLE foo CONFIGURE ZONE USING gc.ttlseconds = 1;")
-	rRand, _ := randutil.NewPseudoRand()
-	writeGarbage := func(from, to int) {
-		for i := from; i < to; i++ {
-			runner.Exec(t, "UPSERT INTO foo VALUES ($1, $2)", i, randutil.RandBytes(rRand, 1<<10))
+
+	for _, testrun := range []struct {
+		name      string
+		runBackup func(t *testing.T, query string, sqlDB *sqlutils.SQLRunner)
+	}{
+		{
+			"backup-normal",
+			func(t *testing.T, query string, sqlDB *sqlutils.SQLRunner) {
+				sqlDB.Exec(t, query)
+			},
+		},
+		{
+			"backup-detached",
+			func(t *testing.T, query string, sqlDB *sqlutils.SQLRunner) {
+				backupWithDetachedOption := query + `WITH DETACHED`
```

Comment:
nit: shouldn't there be a space before "WITH"?

Reply:
done.